unit-tests/utilities/concurrency/test-dispatcher.cpp

```diff
@@ -95,3 +96,40 @@ TEST_CASE("verify stop() not consuming high CPU usage")
     // We allow 9 seconds to support debug configuration as well
     REQUIRE(sw.get_elapsed_ms() < 9000);
 }
+
+TEST_CASE("stop() notify flush to finish")
+{
+    // On this test we check that if during a flush() another thread call stop(),
+    // than the flush CV will be triggered to exit and not wait a full timeout
+    dispatcher dispatcher( 10 );
+    dispatcher.start();
+
+    stopwatch sw;
+    std::atomic_bool dispatched_end_verifier{ false };
+    dispatcher.invoke( [&]( dispatcher::cancellable_timer c ) {
+        //std::cout << "Sleeping from inside invoke" << std::endl;
+        std::this_thread::sleep_for( std::chrono::seconds( 5 ) );
+        //std::cout << "Sleeping from inside invoke - Done" << std::endl;
+        dispatched_end_verifier = true;
+    } );
+
+    // Make sure the above invoke function is dispatched
+    std::this_thread::sleep_for( std::chrono::milliseconds( 100 ) );
```

Comment:
I'm not asking you to do this now unless you feel like it, but:
It's be nice if we could write something like this:
```
    waited_on< bool > started_waiting{ false };
    dispatcher.invoke( [&]( dispatcher::cancellable_timer c ) {
        started_waiting.signal( true );
        std::this_thread::sleep_for( std::chrono::seconds( 5 ) );
        dispatched_end_verifier = true;
    } );

    // Make sure the above invoke function is dispatched
    bool value = started_waiting.wait_until_signalled();
```

Reply:
Or do we already have this in `waiting_on`?

Reply:
We could use a base waiting_on that does not require a predicate for the `wait_until` function and thus not need the user to supply a mutex and CV, and an extended class that does what we do now.

But as we said, we do want to put an extra effort on this currently.